cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -602,6 +787,80 @@ const FunctionDoc pow_checked_doc{
     ("An error is returned when integer to negative integer power is encountered,\n"
      "or integer overflow is encountered."),
     {"base", "exponent"}};
+
+const FunctionDoc sin_doc{"Computes the sine of the elements argument-wise",
+                          ("Integer arguments return double values. "
+                           "This function returns NaN on values outside its domain. "
+                           "To raise an error instead, see sin_checked."),
+                          {"x"}};
```

Comment:
Based on all the other function docs, when a function is referenced in the text it is enclosed in double quotes. For example, change *sin_checked* to *\\"sin_checked\\"*.